idaes/models/control/tests/test_antiwindup.py

```diff
@@ -0,0 +1,571 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Tests to ensure the controller behaves as expected when antiwindup is and isn't
+activated. It turns out that you need to work really hard in order to get
+windup issues in an (approximately) first order system. What to do? Use an
+(approximately) second order system, of course!
+
+           valve_1   +----+                +----+   valve_3
+  steam ----|><|-->--| ta |    valve_2     | ta |----|><|-->--- steam
+                     | nk |-----|><|--->---| nk |
+                     | _1 |                | _2 |
+                     +----+                +----+
+
+
+"""
+
+__author__ = "Douglas Allan, John Eslick"
+
+import pytest
+import numpy as np
+import matplotlib.pyplot as plt
+import pyomo.environ as pyo
+from pyomo.network import Arc
+from idaes.core import FlowsheetBlock, MaterialBalanceType
+from idaes.models.unit_models import Heater, Valve
+from idaes.models.properties import iapws95
+from idaes.core.util.initialization import propagate_state
+from idaes.models.control.controller import (
+    PIDController,
+    ControllerType,
+    ControllerMVBoundType,
+    ControllerAntiwindupType,
+)
+import idaes.core.util.scaling as iscale
+from idaes.core.util.math import safe_sqrt
+from idaes.core.solvers import get_solver, petsc
+import idaes.logger as idaeslog
+from idaes.core.util.plot import plot_grid_dynamic
+import idaes.core.util.scaling as iscale
+
+
+def set_indexed_variable_bounds(var, bounds):
+    for idx, subvar in var.items():
+        subvar.bounds = bounds
+
+
+def _valve_pressure_flow_cb(b):
+    """
+    For vapor F = Cv*sqrt(Pi**2 - Po**2)*f(x)
```

Comment:
The equation in this docstring doesn't match what was implemented. Looks like you were trying a few different things and maybe forgot to update this to match your final version?